buddhi/lib/amp/toolkit/3scale/helper.rb

```diff
@@ -11,6 +15,259 @@ def self.parse_uri(uri)
 
           uri_obj
         end
+
+        def self.create_service(client)
+          service_name = "PERF_TEST_#{random_lowercase_name}"
+          system_name = service_name.delete("\s").downcase
+          svc_params = { 'name' => service_name, 'system_name' => system_name }
+          svc_obj = client.create_service svc_params
+
+          if (errors = svc_obj['errors'])
+            raise "Service has not been created: #{errors}"
+          end
+
+          svc_obj
+        end
+
+        def self.create_application_plan(client, service)
+          name = random_lowercase_name
+          plan_params = {
+            'name' => name, 'state' => 'published', 'default' => false,
+            'custom' => false, 'system_name' => name
+          }
+
+          plan_obj = client.create_application_plan service.fetch('id'), plan_params
+          if (errors = plan_obj['errors'])
+            raise "Application plan has not been created: #{errors}"
+          end
+
+          plan_obj
+        end
+
+        def self.hits_metric(client, service)
+          metrics = client.list_metrics service.fetch('id')
+          if metrics.respond_to?(:has_key?) && (errors = metrics['errors'])
+            raise "Service metrics not read: #{errors}"
+          end
+
+          hits_metric_obj = metrics.find { |metric| metric['system_name'] == 'hits' }
+          raise "Missing hits metric in service #{service.fetch('id')}" if hits_metric_obj.nil?
+
+          hits_metric_obj
+        end
+
+        def self.create_application_plan_limit(client, service, plan)
+          hits_metric_obj = hits_metric(client, service)
+
+          # Very high limit: 4294967295 / 3600 => 1.2 M req/second during one hour to go over limit
+          limit_params = { 'period' => 'hour', 'value' => 2**32 - 1 }
```

Comment:
Why is a very high hourly limit set? Is it done to try to ensure not reaching the limit during the tests? If that's the reason, shouldn't an "eternity" period fit there?

Reply:
an eternity period would be dangerous because eventually, after many tests you could reach the limit and all requests would be rejected. 

With hourly limit, we know that if test load does not exceed 1.2M req/sec, the limit will never be reached, no matter the number of tests you do.

Reply:
Oh, ok.

I thought eternity period meant that there was no real "limit", but now I realize it must mean that the request count is never reset :+1: